src/dataflowd/src/lib.rs

```diff
@@ -145,27 +153,51 @@ impl<S: Client> Client for SplitClient<S> {
 
         Ok(())
     }
-    async fn recv(&mut self) -> Option<Response> {
-        // TODO: We currently don't have a good way to receive from many clients
-        let mut futures = FuturesUnordered::new();
-        for client in self.compute_clients.values_mut() {
-            if let ComputeClientFlavor::Remote(client) = client {
-                futures.push(client.recv());
+}
+
+impl<C: Client> Stream for SplitClient<C> {
+    type Item = Response;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = &mut *self;
+
+        // Poll {each of the compute clients, the storage client} in turn.
+        for _ in 0..=this.compute_clients.len() {
+            match this.compute_clients.get_mut(&this.recv_cursor) {
+                None => {
+                    // We've run out of compute clients. Set up to poll the
+                    // first compute client on the next turn, then poll the
+                    // storage client.
+                    this.recv_cursor = 0;
+                    if let Poll::Ready(res) = Pin::new(&mut this.storage_client).poll_next(cx) {
+                        return Poll::Ready(res);
+                    }
+                }
+                Some(compute_client) => {
+                    // We've found a compute client. Set up to poll the next
+                    // compute client on the next turn, then poll the compute
+                    // client.
+                    this.recv_cursor += 1;
```

Comment:
I think this is where an opaque `ComputeInstanceId` would have stopped you. Maybe we should do that vs a `typedef`? Idk.

Reply:
I’m up for that!